bot/exts/evergreen/quack.py

```diff
@@ -0,0 +1,65 @@
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+
+class Quack(commands.Cog):
+    """Cog for the .quack command."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @commands.command()
+    async def quack(self, ctx: commands.Context, seed: Optional[str], ducktype: Optional[str] = None) -> None:
+        """
+        Pulls a random URL from the Quackstack API.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        If a duck type is provided, either a duck or manduck will be generated.
+        """
+        quackstack_url = "https://quackstack.pythondiscord.com/"
+
+        image_url = "https://quackstack.pythondiscord.com"
+
+        if not ducktype:
+            quackstack_url += "duck"
+        else:
+            if ducktype.lower() not in ["duck", "manduck"]:
+                raise commands.BadArgument(
+                    "The given duck type is invalid. The duck type can either be `duck` or `manduck`."
+                )
+            quackstack_url += ducktype.lower()
+        if seed:
+            try:
+                seed = int(seed)
+            except ValueError:
+                newseed = ""
+                for e in seed:
+                    newseed += str(ord(e))
+                seed = int(newseed)
+            quackstack_url += f"?seed={seed}"
+
+        async with self.bot.http_session.get(quackstack_url) as r:
+            data = await r.json()
+            file = data.get("file")
+            image_url += file
+
+        embed = discord.Embed(
+            title=f"Quack! Here's a {ducktype.lower() if ducktype else 'duck'} for you.",
+            description=f"A {ducktype.lower() if ducktype else 'duck'} from \
+                [Quackstack](https://quackstack.pydis.org/docs).",
```

Comment:
It might be better to link the github page, as it should give a bit more context about what it is, although then again the README isn't really very detailed currently. Maybe it's fine to leave it as this. Thoughts?

Reply:
I think I'm gonna leave it like this for now